proto/engine/v1/execution_engine.go

```diff
@@ -0,0 +1,93 @@
+package enginev1
+
+import "github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+
+type Cloneable[T any] interface {
+	Copy() T
+}
+
+// CopySlice copies the contents of a slice of pointers to a new slice.
+func CopySlice[T any, C Cloneable[T]](original []C) []T {
```

Comment:
maybe I should move this to some util package?

Reply:
IMO should probably be unexported, as it's a trivial func and copying it around seems cleaner to me than making a util that all the proto packages need to import.